src/app/features/date-range/services/controller/controller.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { UntilDestroy, untilDestroyed } from '@app/shared/until-destroy/until-destroy';
+import { filter, of, switchMap, tap } from 'rxjs';
+import { DateRangeApi } from '..';
+import { DateRangeComponent } from '../../components/date-range/date-range.component';
+import { UnavailableDates } from '../../models';
+
+@UntilDestroy
+@Injectable({
+  providedIn: 'root',
+})
+export class ControllerService {
+  constructor(private api: DateRangeApi, private dialog: MatDialog) {}
+
+  getUnavailablePeriods(equipmentId?: number) {
+    this.api
+      .getUnavailablePeriods(equipmentId)
+      .pipe(
+        tap((periods) => this.openDateRangeModal(periods.items, equipmentId)),
+        untilDestroyed(this),
+      )
```

Comment:
we should not have a pipe like this. at this point it does not work as we need because we have only tap. So it is not possible to pass to pipe value from date range component. And this pipe should be in the page component. Page should send a request to get unavalable dates and then open the modal window.

Reply:
done